src/ui/public/fancy_forms/kbn_form_controller.js

```diff
@@ -1,74 +1,81 @@
-import _ from 'lodash';
+export function decorateFormController($delegate, $injector) {
+  const [directive] = $delegate;
+  const FormController = directive.controller;
 
-/**
- * Extension of Angular's FormController class
- * that provides helpers for error handling/validation.
- *
- * @param {$scope} $scope
- */
-export function KbnFormController($scope, $element) {
-  const self = this;
+  class KbnFormController extends FormController {
```

Comment:
> remove unused and unnecessary predicate argument from invalid models getter

Honestly I think `predicate` may have been handy, since `softErrorCount` is called every 1.5s in Visualize view (caused by some `Looper` I think). But I'd agree that is mostly the problem of `ng-if="visualizeEditor.softErrorCount() > 0"` that is used in the markup. We should probably use something smarter there.